Fix E2E test compatibility issues w Woo 10.1

joemcgill opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

See GOOWOO-225.

Fixes some E2E test failures when testing compatibility with Woo 10.1.0-rcX

### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. [Review E2E tests for this branch](https://github.com/woocommerce/google-listings-and-ads/actions/runs/16758648108)

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

Dev – Update E2E tests ahead of Woo 10.1 release.

joemcgill committed:
Fix some race conditions in onboarding step 3 tests

joemcgill committed:
Skip block editor integrations for now

joemcgill commented on tests/e2e/specs/onboarding/step-3-complete-campaign.test.js:
```diff
@@ -692,7 +692,7 @@ test.describe( 'Complete your campaign', () => {
 				await completeCampaign.clickSkipPaidAdsCreationButton();
 				await completeCampaign.clickCompleteSetupModalButton();
-				await page.waitForNavigation( {
+				await page.waitForURL( /path=%2Fgoogle%2Fproduct-feed/, {
```
`page.waitForNavigation()` is deprecated because it can be racy. This updates the test to use `waitForURL` as recommended in [the docs](https://playwright.dev/docs/api/class-page#page-wait-for-navigation).

joemcgill commented on tests/e2e/specs/product-editor/block-integration.test.js:
```diff
@@ -13,6 +13,7 @@ test.use( { storageState: process.env.ADMINSTATE } );
 test.describe.configure( { mode: 'serial' } );
 
 test.describe( 'Product Block Editor integration', () => {
+	test.skip( 'Experimental block editor is deprecated in v10.1.0' );
```
Should we conditionally skip these based on the version of the plugin being tested, or should we remove these altogether for now?

eason9487 replied:
Considering that WooCommerce 10.1 has only removed its feature option from the UI, it can remain as is, with tests skipped but not removed. Future actions could be aligned with WooCommerce Core's deprecation or adjustments to its direction and progress.

joemcgill commented on tests/e2e/specs/onboarding/step-3-complete-campaign.test.js:
```diff
@@ -409,14 +409,14 @@ test.describe( 'Complete your campaign', () => {
 					await completeCampaign.mockCompleteAdsSetup();
 					await completeCampaign.fulfillBudgetRecommendations();
 					await completeCampaign.goto();
+					await page.evaluate( () => window.sessionStorage.clear() );
```
These two `page.evaluate()` calls (see below) were failing due to a race condition with page navigation events when I was testing locally. Any concerns with moving this until after we're on the page state we expect?

eason9487 replied:
I am unable to reproduce the issue you encountered, so I cannot verify whether moving this line is the appropriate solution. It seems counterintuitive here, but I don't think it's a big problem if you confirm it's necessary.

joemcgill replied:
I'm unsure why the order matters in this case, but waiting until after the tests have navigated to the correct screen before clearing sessionStorage seems to make this more consistent.

joemcgill reviewed (commented):
@eason9487 curious if you could take a look at these changes that were flagged as part of compatibility testing of 10.1.0.

I'm unsure if we are better off to remove the block integration tests altogether, or if we should try to conditionally run these depending on the Woo Core version. Everything else seemed to be related to race conditions when page navigations occur.

eason9487 commented on tests/e2e/specs/onboarding/step-3-complete-campaign.test.js:
```diff
@@ -487,10 +487,10 @@ test.describe( 'Complete your campaign', () => {
 			test.describe( 'With WooCommerce tracking disabled', () => {
 				test.beforeAll( async () => {
 					// Reset the showing status for the "Set up paid ads" section.
-					await page.evaluate( () => window.sessionStorage.clear() );
 					await setupAdsAccountPage.mockAdsAccountIncomplete();
 					await completeCampaign.goto();
 					await completeCampaign.clickSkipPaidAdsCreationButton();
+					await page.evaluate( () => window.sessionStorage.clear() );
```
Using `sessionStorage` to conditionally show the campaign section should have been removed by #2533. The code comment "Reset the showing status [...]" and this `window.sessionStorage.clear()` seem to be redundant.

joemcgill replied:
I can confirm that removing the line to run window.sessionStorage.clear() has no affect on the tests passing, so I think it can be safely removed. Handled in d92ad5fcf.

vikrampm1 commented:
@joemcgill @asvinb looks like some review feedback needs to be addressed here before we make the PR ready to be merged?

joemcgill committed:
Remove unnecessary sessionStorage.clear()

joemcgill commented:
@eason9487 I've addressed your comments and confirmed that the E2E tests still pass on both [10.1.rc-2](https://github.com/woocommerce/google-listings-and-ads/actions/runs/16896175908) and the [current version of WC](https://github.com/woocommerce/google-listings-and-ads/actions/runs/16896458967). If you're happy to approve these changes, we can merge this and handle the "tested up to" bump in the next plugin release.

eason9487 reviewed (approved):
Thanks for the work!
